job/server/src/main/java/alluxio/master/job/JobCoordinator.java

```diff
@@ -67,10 +77,13 @@
    */
   private UfsManager mUfsManager;
 
-  private JobCoordinator(CommandManager commandManager, UfsManager ufsManager,
+  private JobCoordinator(CommandManager commandManager, FileSystem filesystem,
```

Comment:
Should this take a `JobMasterContext`?

Reply:
Not in the current architecture - but if we were to redesign the classes and objects in the Job service, then I would say it's a possibility

Reply:
I see what you're getting at how the JobMasterContext should be created once, instead of making many of them. I agree. However, I usually like the keep bug fixes separate from refactors. I can roll both into this PR if you want though

Reply:
Let's get the bug fix in first and then address style/code structure comments.